workspaces/adventure-pack/goodies/typescript/Iterator.prototype.toSet/index.test.ts

```diff
@@ -0,0 +1,113 @@
+import { describe, expect, it } from "@jest/globals";
+
+import "../Iterator.prototype.filter";
+import "../Iterator.prototype.map";
+
+import { iteratorPrototype } from "../Iterator.prototype";
+delete (iteratorPrototype as unknown as Record<string, unknown>).toSet;
+// eslint-disable-next-line import-x/first -- This has to happen after we delete the built-in implementation.
+import "./index";
+
+describe("Iterator.prototype.toSet", () => {
+  it("adds primitives to a Set", () => {
+    const arr = [1, 2, 2, 3, 1, 2, 3, 3, 100, false, 100, false, "hello"];
+    const set = arr.values().toSet();
+
+    expect(set).toStrictEqual(new Set([1, 2, 3, 100, false, "hello"]));
+  });
+
+  it("adds objects to a Set", () => {
+    const obj1 = { a: 1 };
+    const obj2 = { b: 2 };
+    const objectArray = [obj1, obj2, obj1, obj2, obj2, obj1];
+    const set = objectArray.values().toSet();
+
+    expect(set).toStrictEqual(new Set([obj1, obj2]));
+  });
+
+  it("adds arrays to a Set", () => {
+    const arr1 = [1, 2];
+    const arr2 = [3, 4];
+    const arrayOfArrays = [
+      arr1,
+      arr2,
+      arr1,
+      arr1,
+      arr1,
+      arr1,
+      arr1,
+      arr1,
+      arr2,
+      arr2,
+      arr2,
+    ];
+    const set = arrayOfArrays.values().toSet();
+
+    expect(set).toStrictEqual(new Set([arr1, arr2]));
+  });
+
+  it("can convert multiple Generators to a Set", () => {
+    const generator1 = function* (): Generator<string, void, void> {
+      yield "a";
+      yield "b";
+      yield "c";
+      yield "a";
+    };
+
+    const generator2 = function* (): Generator<string, void, void> {
+      yield "x";
+      yield "y";
+      yield "z";
+      yield "x";
+    };
+
+    const combinedIterator = function* (): Generator<string, void, void> {
+      yield* generator1();
+      yield* generator2();
+    };
+
+    const set = combinedIterator().toSet();
+
+    expect(set).toStrictEqual(new Set(["a", "b", "c", "x", "y", "z"]));
+  });
+
+  it("adds falsy values to a Set", () => {
+    const falsyValues = [
+      false,
+      0,
+      "",
+      null,
+      undefined,
+      NaN,
+      false,
+      null,
+      false,
+      NaN,
```

Comment:
Yes, having multiple NaNs is a good idea, since NaN has the interesting distinction of not being equal to itself!

```ts
console.log(NaN === NaN); // will print false
```